File: src/Ccmc.Core/Models/GitInfo.cs
namespace Ccmc.Core.Models;

/// <summary>Lightweight git status for a project row.</summary>
/// <param name="Branch">Current branch name, or a short commit hash when detached.</param>
/// <param name="IsDirty">True when the working tree has uncommitted changes; null when unknown (git CLI unavailable or timed out).</param>
public sealed record GitInfo(string Branch, bool? IsDirty);

File: electron/src/core/models/git-info.ts
// source: GitInfo.cs — in-memory only; camelCase is idiomatic TS
/** Lightweight git status for a project row. */
export interface GitInfo {
  /** Current branch name, or a short commit hash when detached. */
  branch: string;
  /** True when the working tree has uncommitted changes; null when unknown (git CLI unavailable or timed out). */
  isDirty: boolean | null;
}
